apps/bfd-model/bfd-model-rif-samples/src/main/java/gov/cms/bfd/model/rif/samples/StaticRifResource.java

```diff
@@ -19,378 +19,396 @@
       RifFileType.OUTPATIENT,
-      22439),
+      Optional.of(22439)),
 
   SYNTHETIC_OUTPATIENT_2001_1999(
       remoteS3Data(TestDataSetLocation.SYNTHETIC_DATA, "synthetic-outpatient-2001-1999.rif"),
       RifFileType.OUTPATIENT,
-      23241),
+      Optional.of(23241)),
 
   SYNTHETIC_OUTPATIENT_2002_2000(
       remoteS3Data(TestDataSetLocation.SYNTHETIC_DATA, "synthetic-outpatient-2002-2000.rif"),
       RifFileType.OUTPATIENT,
-      24575),
+      Optional.of(24575)),
 
   SYNTHETIC_OUTPATIENT_2014_2014(
       remoteS3Data(TestDataSetLocation.SYNTHETIC_DATA, "synthetic-outpatient-2014-2014.rif"),
       RifFileType.OUTPATIENT,
-      25194),
+      Optional.of(25194)),
 
   SYNTHETIC_OUTPATIENT_2015_2014(
       remoteS3Data(TestDataSetLocation.SYNTHETIC_DATA, "synthetic-outpatient-2015-2014.rif"),
       RifFileType.OUTPATIENT,
-      26996),
+      Optional.of(26996)),
 
   SYNTHETIC_OUTPATIENT_2016_2014(
       remoteS3Data(TestDataSetLocation.SYNTHETIC_DATA, "synthetic-outpatient-2016-2014.rif"),
       RifFileType.OUTPATIENT,
-      27955),
+      Optional.of(27955)),
 
   SAMPLE_MCT_BENES(
-      resourceUrl("rif-static-samples/sample-mct-beneficiaries.txt"), RifFileType.BENEFICIARY, 8),
+      resourceUrl("rif-static-samples/sample-mct-beneficiaries.txt"),
+      RifFileType.BENEFICIARY,
+      Optional.of(8)),
 
-  SAMPLE_MCT_PDE(resourceUrl("rif-static-samples/sample-mct-pde.txt"), RifFileType.PDE, 40),
+  SAMPLE_MCT_PDE(
+      resourceUrl("rif-static-samples/sample-mct-pde.txt"), RifFileType.PDE, Optional.of(40)),
 
   SAMPLE_MCT_UPDATE_1_BENES(
       resourceUrl("rif-static-samples/sample-mct-update-1-beneficiaries.txt"),
       RifFileType.BENEFICIARY,
-      8),
+      Optional.of(8)),
 
   SAMPLE_MCT_UPDATE_2_PDE(
-      resourceUrl("rif-static-samples/sample-mct-update-2-pde.txt"), RifFileType.PDE, 40),
+      resourceUrl("rif-static-samples/sample-mct-update-2-pde.txt"),
+      RifFileType.PDE,
+      Optional.of(40)),
 
   SAMPLE_MCT_UPDATE_3_BENES(
       resourceUrl("rif-static-samples/sample-mct-update-3-beneficiaries.txt"),
       RifFileType.BENEFICIARY,
-      2),
+      Optional.of(2)),
 
   SAMPLE_MCT_UPDATE_3_PDE(
-      resourceUrl("rif-static-samples/sample-mct-update-3-pde.txt"), RifFileType.PDE, 7),
+      resourceUrl("rif-static-samples/sample-mct-update-3-pde.txt"),
+      RifFileType.PDE,
+      Optional.of(7)),
 
   SAMPLE_HICN_MULT_BENES(
       resourceUrl("rif-static-samples/sample-hicn-mult-bene-beneficiaries.txt"),
       RifFileType.BENEFICIARY,
-      10),
+      Optional.of(10)),
 
   SAMPLE_HICN_MULT_BENES_BENEFICIARY_HISTORY(
       resourceUrl("rif-static-samples/sample-hicn-mult-bene-beneficiaryhistory.txt"),
       RifFileType.BENEFICIARY_HISTORY,
-      7);
+      Optional.of(7));
 
   private final Supplier<URL> resourceUrlSupplier;
   private final RifFileType rifFileType;
-  private final int recordCount;
+  private Optional<Integer> recordCount;
```

Comment:
it is not obvious to me why we need an optional here. we should have a comment explaining the conditions where this would be set to an empty value and what the consequences of that are.